geopandas/sindex.py

```diff
@@ -293,27 +280,21 @@ class PyGEOSSTRTreeIndex(pygeos.STRtree):
             GeoSeries from which to build the spatial index.
         """
 
-        # helper for loc/label based indexing in `intersection` method
-        with_objects = namedtuple("with_objects", "object id")
-
         # set of valid predicates for this spatial index
         # by default, the global set
         valid_query_predicates = VALID_QUERY_PREDICATES
 
         def __init__(self, geometry):
-            # for compatibility with old RTree implementation, store ids/indexes
-            original_indexes = geometry.index
             # set empty geometries to None to avoid segfault on GEOS <= 3.6
             # see:
             # https://github.com/pygeos/pygeos/issues/146
             # https://github.com/pygeos/pygeos/issues/147
-            non_empty = geometry.values.data.copy()
+            non_empty = geometry.copy()
             non_empty[pygeos.is_empty(non_empty)] = None
             # set empty geometries to None to mantain indexing
-            self.objects = self.ids = original_indexes
             super().__init__(non_empty)
             # store geometries, including empty geometries for user access
-            self.geometries = geometry.values.data.copy()
+            self.geometries = geometry.copy()
 
         def query(self, geometry, predicate=None, sort=False):
             """Wrapper for pygeos.query.
```

Comment:
Based on changes below, suggest slight edit to docstring:

```
geometry : single PyGEOS or shapely geometry
```

Reply:
Could you clarify _which_ docstring you are referencing?

Reply:
Docstring for `query()` starting on line 300

Reply:
Ah I see, done, thank you for catching this.